debian/rules

```diff
@@ -4,222 +4,11 @@ export DH_VERBOSE=1
-	usr/lib/systemd/network									\
-	usr/lib/systemd/system-generators/systemd-gpt-auto-generator				\
-	usr/lib/systemd/system-preset								\
-	usr/lib/systemd/system/local-fs.target.wants/tmp.mount					\
-	usr/lib/systemd/system/multi-user.target.wants/systemd-logind.service			\
-	usr/lib/systemd/system/sysinit.target.wants/systemd-hwdb-update.service			\
-	usr/lib/systemd/system/sysinit.target.wants/systemd-journal-catalog-update.service	\
-	usr/lib/systemd/system/sysinit.target.wants/systemd-update-done.service			\
-	usr/lib/systemd/system/dbus-org.freedesktop.login1.service				\
-	usr/lib/systemd/system/systemd-hwdb-update.service					\
-	usr/lib/systemd/system/systemd-journal-catalog-update.service				\
-	usr/lib/systemd/system/systemd-logind.service						\
-	usr/lib/systemd/system/systemd-nspawn@.service						\
-	usr/lib/systemd/system/systemd-oomd.service						\
-	usr/lib/systemd/system/systemd-update-done.service					\
-	usr/lib/systemd/system/tmp.mount							\
-	usr/lib/systemd/systemd-logind								\
-	usr/lib/systemd/systemd-oomd								\
-	usr/lib/systemd/systemd-update-done							\
-	usr/lib/systemd/systemd-user-runtime-dir						\
-	usr/lib/systemd/user									\
-	usr/lib/systemd/user-preset								\
-	usr/lib/sysusers.d									\
-	usr/lib/tmpfiles.d/etc.conf								\
-	usr/lib/udev/rules.d/70-power-switch.rules						\
-	usr/lib/udev/rules.d/70-uaccess.rules							\
-	/usr/lib/$(DEB_HOST_MULTIARCH)/security							\
-	/usr/lib/$(DEB_HOST_MULTIARCH)/libpam_misc*						\
-	usr/share/dbus-1/system-services/org.freedesktop.login1.service				\
-	usr/share/dbus-1/system-services/org.freedesktop.oom1.service				\
-	usr/share/dbus-1/system.d/org.freedesktop.login1.conf					\
-	usr/share/dbus-1/system.d/org.freedesktop.oom1.conf					\
-	usr/share/doc										\
-	usr/share/factory									\
-	usr/share/locale									\
-	usr/share/pkgconfig									\
-	usr/share/polkit-1
-
-override_dh_auto_configure:
-	cd vendor/systemd;					\
-	    QUILT_PATCHES=debian/patches quilt push -a --fuzz=0
-
-	# Quilt import patches after the last applied patch.
-	# Because we will apply the patches after they are all imported,
-	# we need to add them reversed order.
-	cd vendor/systemd;						\
-	find ../systemd-patches/ -name "*.patch" -print0 |		\
-	    sort -rz |							\
-	    QUILT_PATCHES=debian/patches xargs -0 -n1 quilt import
-
-	cd vendor/systemd; QUILT_PATCHES=debian/patches quilt push -a --fuzz=0
-
-	sed -i '/"minix"/s/^/\/\//' vendor/systemd/src/test/test-path-util.c
-
-	dh_auto_configure --buildsystem=meson+ninja --sourcedirectory=vendor/systemd -- \
-      -Db_lto=true \
-      -Db_pie=true \
-      -Ddebug=false \
-      -Dstrip=true \
-      -Dsplit-usr=false \
-      -Dsplit-bin=true \
-      -Dntp-servers=ntp.ubuntu.com \
-      -Ddns-servers='' \
-      --libdir=/usr/lib/$(DEB_HOST_MULTIARCH) \
-      -Drootlibdir=/usr/lib/$(DEB_HOST_MULTIARCH) \
-      -Dsysvinit-path='' \
-      -Dsysvrcnd-path='' \
-      -Dsulogin-path='/usr/bin/sh' \
```

Comment:
That is also something to test. I expect that emergency shell is not usable because there is no root password.

Reply:
We could add `Environment=SYSTEMD_SULOGIN_FORCE=1` in a drop-in maybe.

Reply:
emergency mode actually worked just fine when I added bash.

Reply:
I am missing something here on why it should work. Mantic's systemd configuration has `-Dsulogin-path=/sbin/sulogin`. And I do not even see where we copy `sulogin` into the initrd. Or create a symlink to bash.

Reply:
bash however is a huge mamonth.

are you missing busybox-initramfs in the initrd, and symlinks to all the commands it provides?

Reply:
I get:
```
You are in emergency mode. After logging in, type "journalctl -xb" to view
system logs, "systemctl reboot" to reboot, "systemctl default" or "exit"
to boot into default mode.
Failed to execute /sbin/sulogin: No such file or directory
```

Reply:
Maybe we are using different kernel command line. For me if I add
```
console=ttyS0 rd.systemd.debug_shell=ttyS0 dangerous rd.systemd.unit=emergency.service
```
I have a shell.

The interesting thing about `sulogin` is that sometimes I see the error message and sometimes I don't. Anyway, I have included it now, and I still get the shell.

We are not missing busybox bits, it is simply that `debug-shell.service` wants to run bash. I have now removed bash and dash and added an override for debug-shell to make it start sh. Said this, including bash is not a big deal compared to sizes of other things we have in the initramfs, but the more we can save the best.

Reply:
That might have looked like working because you had `rd.systemd.debug_shell=ttyS0`.

But now it works for me. The `SYSTEMD_SULOGIN_FORCE` did not seem needed.

Reply:
... It works for me because of `sulogin`.

Reply:
all i am hering is that to preserve previous behaviour we need to ship symlink from sulogin to sh. Given previously we configured to just call sh instead of sulogin.

Reply:
After making `sulogin` a symbolic link to `/bin/sh` I don't get a shell if using `dangerous rd.systemd.unit=emergency.service`, it just stops execution. So I have kept `sulogin` for the moment.